File: src/main/java/net/liyze/basin/script/commands/StopCommand.java
package net.liyze.basin.script.commands;

import net.liyze.basin.context.annotation.Component;
import net.liyze.basin.core.Basin;
import net.liyze.basin.script.Command;
import org.jetbrains.annotations.NotNull;

import java.util.List;

/**
 * Stop basin after all task finished.
 *
 * @author Liyze09
 */
@Component
public class StopCommand implements Command {
    @Override
    public void run(@NotNull List<String> args) {
        Basin.getBasin().shutdown();
    }

    @Override
    public @NotNull String Name() {
        return "stop";
    }
}

File: src/main/kotlin/net/liyze/basin/core/scan/StopCommand.kt
package net.liyze.basin.core.scan

import net.liyze.basin.core.Command
import net.liyze.basin.core.shutdown

/**
 * Stop basin after all task finished.
 *
 * @author Liyze09
 */

class StopCommand : Command {
    override fun run(args: List<String?>) {
        shutdown()
    }

    override fun Name(): String {
        return "stop"
    }
}
